### PR TITLE
Euler

### PR DESCRIPTION
**Pull request by eggrobin**

### CODE REVIEW

File: geometry/symmetric_bilinear_form_body.hpp [hunk 1]
```diff
@@ -130,6 +130,9 @@ typename SymmetricBilinearForm<Scalar, Frame, Multivector>::
     template Eigensystem<Eigenframe>
     SymmetricBilinearForm<Scalar, Frame, Multivector>::Diagonalize() const {
   R3x3Matrix<Scalar> const& A = matrix_;
+
+  // TODO(phl): This produces NaNs when given a diagonal matrix.
+
```
Comment (pleroy) [inline geometry/symmetric_bilinear_form_body.hpp]: Remove.

File: ksp_plugin/pile_up.cpp [hunk 2]
```diff
@@ -149,14 +156,14 @@ void PileUp::WriteToMessage(not_null<serialization::PileUp*> message) const {
   for (auto const& pair : actual_part_rigid_motion_) {
     auto const part = pair.first;
     auto const& rigid_motion = pair.second;
-    rigid_motion.WriteToMessage(&(
-        (*message->mutable_actual_part_rigid_motion())[part->part_id()]));
+    rigid_motion.WriteToMessage(
+        &((*message->mutable_actual_part_rigid_motion())[part->part_id()]));
```
Comment (pleroy) [inline ksp_plugin/pile_up.cpp]: Many spurious formatting changes.  Please undo.

File: ksp_plugin/pile_up.cpp [hunk 3]
```diff
@@ -295,8 +307,53 @@ PileUp::PileUp(
       psychohistory_(psychohistory),
       deletion_callback_(std::move(deletion_callback)) {}
 
-void PileUp::DeformPileUpIfNeeded() {
+void PileUp::MakeEulerSolver(
+    InertiaTensor<NonRotatingPileUp> const& inertia_tensor,
+    Instant const& t) {
+  LOG(ERROR) << inertia_tensor;
```
Comment (pleroy) [inline ksp_plugin/pile_up.cpp]: Remove the traces.

File: ksp_plugin/pile_up.cpp [hunk 4]
```diff
@@ -295,8 +307,53 @@ PileUp::PileUp(
...
+  auto const eigensystem = inertia_tensor.Diagonalize<PileUpPrincipalAxes>();
+  LOG(ERROR) << eigensystem.form;
```
Comment (pleroy) [inline ksp_plugin/pile_up.cpp]: Same.

File: ksp_plugin/pile_up.cpp [hunk 5]
```diff
@@ -98,7 +101,9 @@ Status PileUp::DeformAndAdvanceTime(Instant const& t) {
   absl::MutexLock l(lock_.get());
   Status status;
   if (psychohistory_->back().time < t) {
-    DeformPileUpIfNeeded();
+    angular_momentum_ += intrinsic_torque_ * (t - psychohistory_->back().time) +
+                         angular_momentum_change_;
```
Comment (pleroy) [inline ksp_plugin/pile_up.cpp]: Could we move this computation to the beginning of `DeformPileUpIfNeeded`?  It's odd that it's floating here for no good reason.

File: ksp_plugin/pile_up.cpp [hunk 6]
```diff
@@ -295,8 +307,53 @@ PileUp::PileUp(
...
+  for (auto const& [part, actual_rigid_motion] : actual_part_rigid_motion_) {
+    rigid_pile_up_.emplace(
+        part,
+        to_pile_up_principal_axes * actual_rigid_motion.rigid_transformation());
+  }
+}
+
+void PileUp::DeformPileUpIfNeeded(Instant const& t) {
   if (apparent_part_rigid_motion_.empty()) {
+    Bivector<AngularMomentum, PileUpPrincipalAxes> angular_momentum =
```
Comment (pleroy) [inline ksp_plugin/pile_up.cpp]: `const`

File: ksp_plugin/pile_up.cpp [hunk 7]
```diff
@@ -295,8 +307,53 @@ PileUp::PileUp(
...
+        euler_solver_->AngularMomentumAt(t);
+    Rotation<PileUpPrincipalAxes, NonRotatingPileUp> const attitude =
+        euler_solver_->AttitudeAt(angular_momentum, t);
+    AngularVelocity<NonRotatingPileUp> angular_velocity_of_rigid_pile_up =
```
Comment (pleroy) [inline ksp_plugin/pile_up.cpp]: `const`

File: ksp_plugin/pile_up.hpp [hunk 8]
```diff
@@ -145,16 +156,16 @@ class PileUp {
          not_null<Ephemeris<Barycentric>*> ephemeris,
          std::function<void()> deletion_callback);
 
+  void MakeEulerSolver(InertiaTensor<NonRotatingPileUp> const& inertia_tensor,
```
Comment (pleroy) [inline ksp_plugin/pile_up.hpp]: Comment this function.

File: ksp_plugin/pile_up.cpp (hunk 5 above)
Comment (eggrobin) [reply ksp_plugin/pile_up.cpp]: That does not make sense.

**Pull request merged**